File: python/mail_items_list.py
import os
import sys
import requests

BASE_URL = os.getenv("PSM_BASE_URL", "https://api.postscanmail.com/api/account-docs/v2")
API_KEY = os.getenv("PSM_API_KEY", "YOUR_API_KEY")

def main():
    url = f"{BASE_URL}/items"
    params = {"sort_order": "desc", "page": 1}
    r = requests.get(url, headers={"x-api-key": API_KEY}, params=params, timeout=30)

    if not r.ok:
        print(f"Request failed: {r.status_code}\n{r.text}", file=sys.stderr)
        sys.exit(1)

    data = r.json()
    print("Mail items fetched successfully.")
    print("Response type:", "list" if isinstance(data, list) else type(data).__name__)

if __name__ == "__main__":
    main()

File: node/mail-items-list.js
const { BASE_URL, API_KEY } = (() => {
  const BASE_URL = process.env.PSM_BASE_URL || "https://api.postscanmail.com/api/account-docs/v2";
  const API_KEY = process.env.PSM_API_KEY || "YOUR_API_KEY";
  return { BASE_URL, API_KEY };
})();

async function main() {
  const url = new URL(`${BASE_URL}/items`);
  url.searchParams.set("sort_order", "desc");
  url.searchParams.set("page", "1");

  const res = await fetch(url.toString(), {
    method: "GET",
    headers: { "x-api-key": API_KEY, "Content-Type": "application/json" },
  });

  const text = await res.text();
  if (!res.ok) {
    console.error(`Request failed: ${res.status}\n${text}`);
    process.exit(1);
  }

  console.log("Mail items fetched successfully.");
  console.log(text);
}

main().catch((err) => {
  console.error("Unexpected error:", err);
  process.exit(1);
});
